src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java

```diff
@@ -85,7 +98,8 @@ public DefaultRecoveryScheduler(
             recoveryRequirement = Optional.of(recoveryCandidates.get(new Random().nextInt(recoveryCandidates.size())));
         }
 
-        if (recoveryRequirement.isPresent() && launchConstrainer.canLaunch(recoveryRequirement.get())) {
+        boolean b = recoveryRequirement.isPresent() && launchConstrainer.canLaunch(recoveryRequirement.get());
```

Comment:
Why does the `b` variable exist?  Shouldn't this all be part of the if statement below?